client/wp-admin-scripts/print-shipping-label-banner/shipping-banner/index.js

```diff
@@ -300,40 +300,61 @@ export class ShippingBanner extends Component {
 
 			const wcsStoreUnsubscribe = wcsStore.subscribe( () => {
 				const latestState = wcsStore.getState();
+
 				const siteState =
 					latestState.extensions.woocommerce.woocommerceServices[
 						siteId
 					];
 				if ( siteState ) {
 					const orderState = siteState.shippingLabel[ orderId ];
 					if ( orderState ) {
-						if ( orderState.showPurchaseDialog ) {
+						if (
+							! orderState.showPurchaseDialog &&
+							! orderState.isFetching
+						) {
```

Comment:
We open the modal only if the modal is not already open, and it the modal is down getting data from WC Connect Server.